src/vmm/src/devices/virtio/net/tap.rs

```diff
@@ -190,6 +190,20 @@ impl Tap {
         }
         Ok(usize::try_from(ret).unwrap())
     }
+
+    /// Read from tap to an `IoVecBufferMut`
+    pub(crate) fn read_iovec(&mut self, buffer: &mut IoVecBufferMut) -> Result<usize, IoError> {
+        let iovcnt = buffer.iovec_count() as i32;
+        let iov = buffer.as_iovec_ptr();
```

Comment:
Here is the issue. When we want to read from TAP, we want to read onto a single chain, but here you give it all `iovec`s, so `readv` can write to all the chains it wants.
To fix this we just need smth like this: https://github.com/firecracker-microvm/firecracker/pull/4813/files#diff-205f89a698c2fafa14dcbd3c76dfee31dd597294c8334bc6b82d8c92814a497fR108

Reply:
Sure, but this won't happen if we make sure that we only keep the descriptor chains that are of the appropriate size. However, I like the extra guarantee, so I'll do this.